working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
+    The expression `augmented` must not occur in an optional parameter
+    default value expressions _It is a reserved word in such expressions,
+    but is currently cannot be used._
+    The term <code>'augmented' \<argumentPart\></code> words parses
+    as a `<primary>` expression.
+    Evaluating the function invocation, after having evaluated the
+    argument list, invokes the augmented function with the provided
+    type arguments and evaluated argument list, and evaluates to the
+    returned value.
+    Type inference of <Code>augmented\<_typeArgs_\>(_args_)<code>
+    is performed as a function invocation with the function signature
+    of the augmented declaration.
+
+*   **Augmenting operators**: Operators are methods with special
+    invocation syntax. When augmenting an operator declaration, `augmented`
+    refers to the augmented operator *method*, and it must occur only
+    as the function of a function invocation. For example, when augmenting
     `operator +` you could use `augmented(1)` to call the augmented
-    operator, and when augmenting `operator []=` you would use the
-    `augmented(key, value)` syntax.
-
-    *   Note that `augmented` in such an augmenting operator method body is
-        not an expression by itself, and cannot be used to tear off the
-        augmented operator method. Similar to `super`, it is a syntactic
-        form which can only be used in limited ways.
+    operator, when augmenting `operator []=` you would use the
+    `augmented(key, value)` syntax, and when augmenting `operator -()`,
+    you would use `augmented()`.
+    Parsing, evaluation and type inference is handled like a
+    function augmentation.
 
 *   **Augmenting enum values**: When augmenting an enum value, `augmented`
-    has no meaning and is not allowed.
+    has no meaning and is not allowed to occur in constructor argument
+    expressions.
+
+*   **Augmenting constructors**:
+    *   **Non-redirecting generative constructor**:
+        The term `augmented` has no meaning, and must not occur in
+        optional parameter default value expressions, initializer list
+        entry expressions (whether initializing or asserts),
+        super-constructor invocation argument expressions, or in the
+        body. _Non-redirecting generative constructors automatically
+        invoke their augmented constructor after executing their
+        initializer list, and will execute their body after that of
+        their augmented constructor._
+    *   **Non-redirecting Factory constructor**: An `augmented` in the
+        constructor body refers to the augmented non-redirecting factory
+        constructor. It must occur only as the function of a (non-generic)
+        function invocation.
+        The term <code>'augmented' \<argumentList\></code> clause parses
+        as a `<primary>` expression.
+        When evaluated, it invokes the augmented constructor on the same
+        class and evaluates to the returned value.
```

Comment:
Maybe add "with the given arguments" just to make it clear that there is no implicit argument forwarding (since generative constructors *do* have the arguments implicitly forwarded).